lib/redis_memo/memoize_method.rb

```diff
@@ -81,14 +81,28 @@ def self.extract_dependencies(ref, *method_args, &depends_on)
 
   def self.get_or_extract_dependencies(ref, *method_args, &depends_on)
     if RedisMemo::Cache.local_dependency_cache
-      RedisMemo::Cache.local_dependency_cache[ref] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on] ||= {}
-      RedisMemo::Cache.local_dependency_cache[ref][depends_on][method_args] ||= extract_dependencies(ref, *method_args, &depends_on)
+      RedisMemo::Cache.local_dependency_cache[ref.class] ||= {}
+      RedisMemo::Cache.local_dependency_cache[ref.class][depends_on] ||= {}
+      mapped_args = map_depends_on_method_args(depends_on, method_args)
+      RedisMemo::Cache.local_dependency_cache[ref.class][depends_on][mapped_args] ||= extract_dependencies(ref, *method_args, &depends_on)
     else
       extract_dependencies(ref, *method_args, &depends_on)
     end
   end
 
+  def self.map_depends_on_method_args(depends_on, args)
+    mapped_args = {}
+    unless depends_on.parameters.empty? or args.empty?
+      params = depends_on.parameters.drop(1)
```

Comment:
hmm, so now we're throwing away the object reference from the `local_dependency_cache` right?

I think this could be an issue tho, because it's also possible for the `depends_on` block to depend on some instance variables

Reply:
Ah you're right, I don't know why I did this 😅